src/raylib/textures/Image.hpp

```diff
@@ -0,0 +1,147 @@
+            ///
+            /// @param rec The rectangle to get the image from.
+            ///
+            /// @return A new image object.
+            Image imageFromImage(Rectangle rec);
+
+            /// It changes the format of the image.
+            ///
+            /// @param newFormat The new format for the image.
+            void imageFormat(int newFormat);
+
+            /// It resizes the image to the new width and height.
+            ///
+            /// @param newWidth The new width of the image.
+            /// @param newHeight The new height of the image.
+            void imageResize(int newWidth, int newHeight);
+
+            /// Flips the image vertically
+            void imageFlipVertical();
+
+            /// Flips the image horizontally
+            void imageFlipHorizontal();
+
+            /// This function rotates the image clockwise by 90 degrees
+            void imageRotateCW();
+
+            /// It rotates the image counter clockwise.
+            void imageRotateCCW();
+
+            /// Tinting the image with the color passed in.
+            /// @param color
+            void imageColorTint(Color color);
+
+            /// It inverts the colors of the image.
+            void imageColorInvert();
+
+            /// This function converts an image to grayscale
+            void imageColorGrayscale();
+
+            /// This function changes the contrast of an image.
+            ///
+            /// @param contrast The contrast to apply to the image.
+            void imageColorContrast(float contrast);
+
+            /// It changes the brightness of the image.
+            ///
+            /// @param image The image to modify
+            /// @param brightness The brightness value.
+            void imageColorBrightness(Image *image, int brightness);
+
+            /// Replaces all instances of a color in an image with another color.
+            ///
+            /// @param color The color to replace.
+            /// @param replace The color to replace
+            void imageColorReplace(Color color, Color replace);
+
+            /// This function loads the color of the image
+            ///
+            /// @param color The color of the image.
+            void loadImageColors(Color *color);
+
+            /// UnloadImageColors() unloads the image colors from the CPU memory (RAM)
+            void unloadImageColors();
+
+            /// It returns the color of the pixel at the given coordinates.
+            ///
+            /// @param image The image to get the color from.
+            /// @param x The x coordinate of the pixel to get the color of.
+            /// @param y The y coordinate of the pixel to get the color of.
+            ///
+            /// @return A Color object.
+            Color getImageColor(Image image, int x, int y);
+
+            /// This function returns the color of the image
+            ///
+            /// @return The color of the image.
+            Color getColor();
```

Comment:
What is the color of the image ? 
The mean color ?
Or should it be an array of colors ?